ksql-execution/src/main/java/io/confluent/ksql/execution/json/PlanJsonMapper.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.execution.json;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+
+public final class PlanJsonMapper {
+  private PlanJsonMapper() {
+  }
+
+  /**
+   * Create an ObjectMapper configured for serializing/deserializing a KSQL physical plan.
+   *
+   * @return ObjectMapper instance
+   */
+  public static ObjectMapper create() {
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModules(
+        new Jdk8Module(),
+        new JavaTimeModule()
+    );
+    mapper.enable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
```

Comment:
Do we really want to fail on unknown properties?  This makes it virtually impossible to evolve the schema in a backwards compatible way, as adding a new field causes old versions to fail...

Reply:
> adding a new field causes old versions to fail

This is what we want. If there's an unrecognized field, that means that some newer version wrote that field and will probably use it when building the final query plan. We therefore don't want the current version to build the plan, because it likely doesn't have enough information to build it correctly. The goal here is to be backwards-compatible. New versions should be able to read and execute old plans, which this permits as long as we only add properties to the schema (we should extend our schema generation/checking to validate this - I'm looking into options here).